### PR TITLE
Move processsnapshotting to um

### PR DESCRIPTION
**Pull request by kichristensen**

### CODE REVIEW

**Comment by retep998** [comment]
Make sure you also reformat the functions to follow the indentation and linebreak guidelines: https://github.com/retep998/winapi-rs/blob/dev/CONTRIBUTING.md#extern-functions

**Comment by kichristensen** [comment]
@retep998 Done

**Comment by Susurrus** [comment]
@kichristensen These function declarations don't follow the same order as the header file `ProcessSnapshot.h`. Please reorder them such that the `PssWalkMarker*()` functions are after `PSS_ALLOCATOR` and the other ones are before, following the same order they are listed in the header file.

**Comment by kichristensen** [comment]
@Susurrus functions should be reordered now

**Comment by retep998** [comment]
You appear to have some duplicate functions causing build failures.

**Comment by Susurrus** [comment]
@kichristensen Can you rebase and update this PR?

File: build.rs [hunk 1]
```diff
@@ -156,7 +156,7 @@ const DATA: &'static [(&'static str, &'static [&'static str], &'static [&'static
     ("olectl", &["winerror", "winnt"], &[]),
     ("pdh", &["basetsd", "guiddef", "minwindef", "windef", "winnt", "wtypesbase"], &["pdh"]),
     ("playsoundapi", &["minwindef"], &[]),
-    ("processsnapshot", &["minwindef"], &[]),
+    ("processsnapshot", &["basetsd", "minwindef", "winnt"], &[]),
```
Comment (retep998) [inline build.rs]: The functions in this header need the library `"kernel32"`.

File: src/um/processsnapshot.rs [hunk 2]
```diff
@@ -51,6 +53,40 @@ ENUM!{enum PSS_DUPLICATE_FLAGS {
 }}
 DECLARE_HANDLE!(HPSS, HPSS__);
 DECLARE_HANDLE!(HPSSWALK, HPSSWALK__);
+
```
Comment (Susurrus) [inline src/um/processsnapshot.rs]: There shouldn't be any blank lines in the Rust source files. This patch adds 3 empty lines.

**Comment by Susurrus** [comment]
@kichristensen This needs a rebase and to address the blank lines in the Rust files.

**Comment by kichristensen** [comment]
@Susurrus @retep998 this is hopefully fixed now

**Comment by Susurrus** [comment]
LGTM

File: src/um/processsnapshot.rs [hunk 3]
```diff
@@ -64,3 +66,56 @@ STRUCT!{struct PSS_ALLOCATOR {
...
+    ) -> DWORD;
+    // pub fn PssWalkMarkerTell();
+    pub fn PssWalkSnapshot(
+        SnapshotHandle: HPSS,
+        InformationClass: PSS_WALK_INFORMATION_CLASS,
+        WalkMarkerHandle: HPSSWALK,
+        Buffer: *mut c_void,
+        BufferLength: DWORD,
+    ) -> DWORD;
+}
```
Comment (retep998) [inline src/um/processsnapshot.rs]: End of file newline!

Comment (kichristensen) [reply src/um/processsnapshot.rs]: Done

**Pull request merged**